conda/cli/common.py

```diff
@@ -165,19 +165,6 @@ def stdout_json(d):
 
 def stdout_json_success(success=True, **kwargs):
     result = {'success': success}
-
-    # this code reverts json output for plan back to previous behavior
-    #   relied on by Anaconda Navigator and nb_conda
-    unlink_link_transaction = kwargs.pop('unlink_link_transaction', None)
-    if unlink_link_transaction:
-        from .._vendor.toolz.itertoolz import concat
-        actions = kwargs.setdefault('actions', {})
-        actions['LINK'] = tuple(d.dist_str() for d in concat(
-            stp.link_precs for stp in itervalues(unlink_link_transaction.prefix_setups)
-        ))
-        actions['UNLINK'] = tuple(d.dist_str() for d in concat(
-            stp.unlink_precs for stp in itervalues(unlink_link_transaction.prefix_setups)
-        ))
```

Comment:
This code was all duplicated logic that meant `--dry-run` had different behavior than without the flag.  Consolidated it now.  Also added tests for both `--dry-run` and without.